src/Psl/DateTime/Duration.php

```diff
@@ -0,0 +1,729 @@
+     * the provided one, and the resulting instance may contain larger units.
+     *
+     * For example, `Duration::minutes(63)` normalizes to "1 hour(s), 3 minute(s)".
+     *
+     * @pure
+     */
+    public static function minutes(int $minutes): self
+    {
+        return self::fromParts(0, $minutes);
+    }
+
+    /**
+     * Returns an instance representing the specified number of seconds. Due to
+     * normalization, the actual value in the returned instance may differ from
+     * the provided one, and the resulting instance may contain larger units.
+     *
+     * For example, `Duration::seconds(63)` normalizes to "1 minute(s), 3 second(s)".
+     *
+     * @pure
+     */
+    public static function seconds(int $seconds): self
+    {
+        return self::fromParts(0, 0, $seconds);
+    }
+
+    /**
+     * Returns an instance representing the specified number of milliseconds (ms).
+     * The value is converted and stored as nanoseconds, since that is the only
+     * unit smaller than a second that we support. Due to normalization, the
+     * resulting instance may contain larger units.
+     *
+     * For example, `Duration::milliseconds(8042)` normalizes to "8 second(s), 42000000 nanosecond(s)".
+     *
+     * @pure
+     */
+    public static function milliseconds(int $milliseconds): self
+    {
+        return self::fromParts(0, 0, 0, NANOSECONDS_PER_MILLISECOND * $milliseconds);
+    }
+
+    /**
+     * Returns an instance representing the specified number of microseconds (us).
+     * The value is converted and stored as nanoseconds, since that is the only
+     * unit smaller than a second that we support. Due to normalization, the
+     * resulting instance may contain larger units.
+     *
+     * For example, `Duration::microseconds(8000042)` normalizes to "8 second(s), 42000 nanosecond(s)".
+     *
+     * @pure
+     */
+    public static function microseconds(int $microseconds): self
+    {
+        return self::fromParts(0, 0, 0, NANOSECONDS_PER_MICROSECOND * $microseconds);
+    }
+
+    /**
+     * Returns an instance representing the specified number of nanoseconds (ns).
+     * Due to normalization, the resulting instance may contain larger units.
+     *
+     * For example, `Duration::nanoseconds(8000000042)` normalizes to "8 second(s), 42 nanosecond(s)".
+     *
+     * @pure
+     */
+    public static function nanoseconds(int $nanoseconds): self
+    {
+        return self::fromParts(0, 0, 0, $nanoseconds);
+    }
+
+    /**
+     * Returns an instance with all parts equal to 0.
+     *
+     * @pure
+     */
+    public static function zero(): self
+    {
+        return new self(0, 0, 0, 0);
+    }
+
+    /**
+     * Compiles and returns the duration's components (hours, minutes, seconds, nanoseconds) in an
+     * array, in descending order of significance.
+     *
+     * @return array{int, int, int, int}
+     *
+     * @psalm-mutation-free
+     */
+    public function getParts(): array
```

Comment:
I think parts should go to years here.

Reply:
getParts and fromParts here tend to reflect the properties stored internally, we could have a function to return the number of days maybe, but since it is not an internal property, it shouldn't be returned here.